### ISSUE TITLE
SourceMap bug about inconsistent path processing

### ISSUE THREAD

**Issue by creeperyang** [webpack/webpack#2616]
Background: A normal project compiled successfully

``` bash
/xxx/proj
       |------ src/
       |------ webpack.config.js
```

And then I move the project source file `/xxx/proj/src` to `/yyy/src`, the compilation will run into error:

``` bash
ERROR in scripts/app.84b4e.js from UglifyJs
Error: "/yyy/src/app.js" is not in the SourceMap.
...
```

The error resulted from `UglifyJsPlugin`, disable it will not get error any more. I dive into the compilation and found the error is about **inconsistent path processing**:

```
// Source we had
{ '$../../../../../yyy/src/app.js': 0 }
// Plugin try to access by
'$/yyy/src/app.js'
```

In fact we have the source map, but plugin try to access via a different path.

And I found the compilation failed at `78% additional chunk assets`. So any help?

**Comment by creeperyang** [comment]
I finally found the error resulted from `babel-loader` and fix it with a pr to `babel-loader`.

`webpack/node_modules/webpack-core/lib/NormalModuleMixin.js`'s `runSyncOrAsync` will run `babel-loader`, and `babel-loader` call:

```js
this.callback(null, result.code, result.map);
/*
    result.map is like:

    { 
      version: 3,
      sources: [ '/private/tmp/wow/viewer/pages/index/viewer.js' ],
      names: [],
      mappings: 'xxxxxxxxx',
      file: 'viewer.js',
      sourceRoot: '/Users/creeper/work/projects/SugarDevelop',
      sourcesContent: [ 'xxxxxxx']
    }
*/

// --->
nextLoader.apply(null, arguments); // arguments is [result.code, result.map]

// --->
onModuleBuild.apply(module, args); // args is [result.code, result.map] 

// --->
this._source = new SourceMapSource(source, this.identifier(), sourceMap);

// and finally we adopt the sources generated by `babel-loader` and run into error
```

**Comment by NickStefan** [comment]
was said PR ever taken by babel-loader @creeperyang  ?

**Comment by danez** [comment]
This is now fixed right with babel-loader 6.2.8? @creeperyang

**Comment by creeperyang** [comment]
Yeah, it's fixed.
